pkg/sql/plan/function/builtin/multi/external.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package multi
+
+import (
+	"github.com/matrixorigin/matrixone/pkg/container/nulls"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/vectorize/external"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+)
+
+type tp1 interface {
```

Comment:
it's better to rename : tp1 -> tpNumber;   tp2 -> tpTime;   tp3 -> tpDecimal
as the same you can rename:
ParseCol1-> ParseNumber ,   ParseCol2-> ParseTime,   ParseCol3->ParseDecimal

Reply:
Fine，I will rename them.